### PR TITLE
Update data transformer docs

### PR DESCRIPTION
**Pull request by cptartur**
Closes #

## Introduced changes

- Separated data transformer guide into two subsections
- Changed `warning` to `attention`

##

- [ ] This PR contains breaking changes

### CODE REVIEW

**Review by war-in — APPROVED** [review]
Can be merged after updating a comment

**Review by war-in — APPROVED** [review]

File: starknet_py/tests/e2e/docs/guide/test_serializing.py [hunk 1]
```diff
@@ -4,16 +4,28 @@
...
+    # docs-shortstring: start
+    from starknet_py.cairo.felt import decode_shortstring, encode_shortstring
+
+    # Convert a string literal to its felt value
+    encoded = encode_shortstring("myshortstring")
+    assert encoded == 0x6D7973686F7274737472696E67
+
+    # Decode a felt into a string
+    # Note that decoded value always has 31 characters in length.
+    # Extra `\x00` characters can be stripped to get only the encoded value.
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/guide/test_serializing.py:17]: Stripped from the end maybe?

File: docs/guide/serialization.rst [hunk 2]
```diff
@@ -85,6 +92,22 @@ for more details.
...
+.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_serializing.py
+    :language: python
+    :dedent: 4
+    :start-after: docs-serializer: start
+    :end-before: docs-serializer: end
+
+Serializing events
+^^^^^^^^^^^^^^^^^^
+
+Events emitted by contracts can also be serialized provided the correct ABI
```
Comment (drknzz) [inline docs/guide/serialization.rst:107]:
```suggestion
Events emitted by contracts can also be serialized, having provided the correct ABI
```

File: starknet_py/tests/e2e/docs/guide/test_serializing.py (hunk 1 above)
Comment (cptartur) [reply starknet_py/tests/e2e/docs/guide/test_serializing.py:17]: Actually from the beginning. Updated the snippet used here so it shows how to strip `\x00`.

**Pull request merged**